Add Salesforce doc (#8012)
fishtown-analytics/docs.getdbt.com: website/docs/docs/fusion/connect-data-platform-fusion/salesforce-data-cloud-setup.md

id: "salesforce-data-cloud-setup"
meta:
  maintained_by: dbt Labs
  authors: 'Fusion dbt maintainers'
  github_repo: 'dbt-labs/dbt-fusion'
  pypi_package: N/A
  min_core_version: N/A
  cloud_support: N/A
  min_supported_version: 'n/a'
  slack_channel_name: N/A
  slack_channel_link: N/A
  platform_name: 'Salesforec Data Cloud'
  config_page: '/reference/resource-configs/data-cloud-configs'
---

:::warning Disclaimer
This adapter is in the Alpha product stage and is not production-ready. It should only be used in sandbox or test environments. 

As we continue to develop and take in your feedback, the experience is subject to change &mdash; commands, configuration, and workflows may be updated or removed in future releases. 
:::

This `dbt-salesforce` adapter is available via CLI in the <Constant name="fusion_engine" />. To access the adapter, [install dbt Fusion](/docs/fusion/about-fusion-install). We recommend using the [VS Code Extension](/docs/fusion/install-dbt-extension) as the development interface. <Constant name="dbt_platform" /> support coming soon. 

## Prerequisites

Before you can connect dbt to the Salesforce Data Cloud, you need the following:

- A Data Cloud instance

> - [An external client app that dbt connects to for the Data Cloud instance](https://help.salesforce.com/s/articleView?id=xcloud.create_a_local_external_client_app.htm&type=5), with [OAuth configured](https://help.salesforce.com/s/articleView?id=xcloud.configure_external_client_app_oauth_settings.htm&type=5)

Reviewer: ```suggestion
- [An external client app that dbt connects to for the Data Cloud instance](https://help.salesforce.com/s/articleView?id=xcloud.create_a_local_external_client_app.htm&type=5), with [OAuth configured](https://help.salesforce.com/s/articleView?id=xcloud.configure_external_client_app_oauth_settings.htm&type=5). Oauth scopes must include:
    - Manage user data via APIs (`api`)
    - Perform requests at any time (`refresh_token`, `offline_access`)
    - Perform ANSI SQL queries on Data Cloud data (`cdp_query_api`)
```

- [A private key and the `server.key` file](https://developer.salesforce.com/docs/atlas.en-us.252.0.sfdx_dev.meta/sfdx_dev/sfdx_dev_auth_key_and_cert.htm)

> - User with `Data Cloud admin` permission set

Reviewer: ```suggestion
- User with `Data Cloud Admin` permission set
```

---

:::warning Disclaimer
This adapter is in the Alpha product stage and is not production-ready. It should only be used in sandbox or test environments. 

As we continue to develop and take in your feedback, the experience is subject to change &mdash; commands, configuration, and workflows may be updated or removed in future releases. 
:::

This `dbt-salesforce` adapter is available via CLI in the <Constant name="fusion_engine" />. To access the adapter, [install dbt Fusion](/docs/fusion/about-fusion-install). We recommend using the [VS Code Extension](/docs/fusion/install-dbt-extension) as the development interface. <Constant name="dbt_platform" /> support coming soon. 

## Prerequisites

Before you can connect dbt to the Salesforce Data Cloud, you need the following:

- A Data Cloud instance
- [An external client app that dbt connects to for the Data Cloud instance](https://help.salesforce.com/s/articleView?id=xcloud.create_a_local_external_client_app.htm&type=5), with [OAuth configured](https://help.salesforce.com/s/articleView?id=xcloud.configure_external_client_app_oauth_settings.htm&type=5). OAuth scopes must include:

>   - Manage user data via APIs (`api`)

Author: ```suggestion
  - `api` - To manage user data via APIs.
```

>   - Perform requests at any time (`refresh_token`, `offline_access`)

Author: ```suggestion
  - `refresh_token`, `offline_access` - To perform requests at any time, even when the user is offline or tokens have expired.
```

>   - Perform ANSI SQL queries on Data Cloud data (`cdp_query_api`)

Author: ```suggestion
  - `cdp_query_api` - To execute ANSI SQL queries on Data Cloud data.
```
